File: app/src/main/java/alejandromr92/com/movies/network/model/converter/MovieNetworkConverter.java
package alejandromr92.com.movies.network.model.converter;

import alejandromr92.com.movies.network.model.response.MovieResponse;
import alejandromr92.com.movies.network.service.Endpoints;
import alejandromr92.com.movies.storage.model.MovieDataStore;

import java.util.ArrayList;
import java.util.List;

public abstract class MovieNetworkConverter {
    public static List<MovieDataStore> convertListToStoreModel(List<MovieResponse> networkList){
        List<MovieDataStore> storeList = new ArrayList<>();

        if (networkList != null && networkList.size() > 0){
            for (MovieResponse m: networkList){
                storeList.add(convertToStoreModel(m));
            }
        }

        return storeList;
    }

    public static MovieDataStore convertToStoreModel(MovieResponse network){
        MovieDataStore store = new MovieDataStore();

        store.setTitle(network.getTitle());
        store.setOverview(network.getOverview());
        store.setPictureUrl(Endpoints.MOVIES_POSTERS_BASE_URL + network.getPictureUrl());
        store.setYear(network.getReleaseDate().substring(0, 4));

        return store;
    }
}

File: app/src/main/java/alejandromr92/com/movies/network/model/converter/MovieNetworkConverter.kt
package alejandromr92.com.movies.network.model.converter

import alejandromr92.com.movies.network.model.response.MovieResponse
import alejandromr92.com.movies.network.service.Endpoints
import alejandromr92.com.movies.storage.model.MovieDataStore
import java.util.ArrayList

abstract class MovieNetworkConverter {
    companion object {
        fun convertListToStoreModel(networkList: List<MovieResponse>?): List<MovieDataStore> {
            val storeList = ArrayList<MovieDataStore>()

            if (networkList != null && networkList.isNotEmpty()) {
                for (m in networkList) {
                    storeList.add(convertToStoreModel(m))
                }
            }

            return storeList
        }

        private fun convertToStoreModel(network: MovieResponse): MovieDataStore {
            val store = MovieDataStore()

            store.title = network.title
            store.overview = network.overview
            store.pictureUrl = Endpoints.MOVIES_POSTERS_BASE_URL + network.poster_path
            store.year = network.release_date.substring(0, 4)

            return store
        }
    }
}